android/src/main/kotlin/com/pusher/channels_flutter/PusherChannelsFlutterPlugin.kt

```diff
@@ -229,12 +229,25 @@ class PusherChannelsFlutterPlugin : FlutterPlugin, MethodCallHandler, ActivityAw
 
     override fun onEvent(event: PusherEvent) {
         // Log.i(TAG, "Received event with data: $event")
+
+        // The java sdk transforms some events from pusher_internal
+        // to pusher:... events, we translate them back.
+        val finalEvent = if (event.eventName === "pusher:subscription_count") {
+            PusherEvent(
+                "pusher_internal:subscription_count",
+                event.channelName,
+                event.userId,
+                event.data)
+        } else {
+            event
+        }
+
         callback(
             "onEvent", mapOf(
-                "channelName" to event.channelName,
-                "eventName" to event.eventName,
-                "userId" to event.userId,
-                "data" to event.data
+                "channelName" to finalEvent.channelName,
+                "eventName" to finalEvent.eventName,
+                "userId" to finalEvent.userId,
+                "data" to finalEvent.data
```

Comment:
Asking because I lack the context: why is this needed?
Looking at `pusher-js`, we seem to the same as the java SDK.

Reply:
We transform them from `pusher_internal` to `pusher` and the second one is the one that people would `bind` to, if I understand correctly. What I don't get is why are explicitly undoing that behaviour for this particular client.